app/src/lib/ai/types.ts

```diff
@@ -39,11 +39,11 @@ export interface AIClient {
 	defaultCommitTemplate: Prompt;
 }
 
-export interface UserPrompt {
+export type UserPrompt = {
 	id: string;
 	name: string;
 	prompt: Prompt;
-}
+};
```

Comment:
Is there a reason to prefer a type over an interface here?

Reply:
In this specific commit, just so that the shape can be inferred into the `SelectItemType extends Record<string, unknown>` generic. Interfaces can't be inferred as something that extends `Record`s

In general, just because `type` will raise compile errors if there are any other types in the same scope. While `interface`s will just merge both by default, which is not always what one wants.
It's not that common so it's not necessary a huge improvement.

Reply:
Reference 😊 https://github.com/microsoft/TypeScript/issues/15300#issuecomment-332366024